docs/berkeley-upgrade/mainnet-database-maintenance.mdx

```diff
@@ -0,0 +1,186 @@
+However, instead of skipping only blocks with bad ledger hashes, this mode skipped all of the problems with integrity.
+With the new archive node maintenance package, you can run the replayer application without a special flag and to correctly handle the bad ledger hashes issue.
+
+To run replayer:
+
+```sh
+mina-replayer --archive-uri {db_connection_string} --input-file reference_replayer_input.json --output-file reference_replayer_output.json --checkpoint-interval 10000
+```
+
+where:
+
+- `archive-uri` - connection string to the archive database 
+- `input-file` - JSON file that holds the archive database
+- `output-file` - JSON file that will hold the ledger with auxiliary information, like global slot and blockchain height, which will be dumped on the last block
+- `checkpoint-interval` - frequency of checkpoints expressed in blocks count
+- `replayer_input_file.json` - JSON file constructed from the Devnet/Mainnet genesis ledger:
+  
+   ```
+      jq '.ledger.accounts' genesis_ledger.json | jq  '{genesis_ledger: {accounts: .}}' > replayer_input_config.json
+   ```
+
+:warning: Running a replayer from scratch on a Devnet/Mainnet database can take up to a couple of days. The recommended best practice is to break the replayer into smaller parts by using the checkpoint capabilities of the replayer.
+
+:warning: You must run the replayer using the Mainnet version. You can run it from the Docker image at `minaprotocol/mina-archive:1.4.0-c980ba8-bullseye`.
+
+## Missing blocks
+
+The daemon node unavailability can cause the archive node to miss some of the blocks. This recurring missing blocks issue consistently poses challenges. To address this issue, you can reapply missing blocks.
+
+If you uploaded the missing blocks to Google Cloud, the missing blocks can be reapplied from precomputed blocks to preserve chain continuity. 
+
+1. To automatically verify and patch missing blocks, use the [download_missing_blocks.sh](https://raw.githubusercontent.com/MinaProtocol/mina/2.0.0berkeley_rc1/src/app/rosetta/download-missing-blocks.sh) script. 
+
+   The `download-missing-blocks` script uses `localhost` as the database host so the script assumes that psql is running on localhost on port 5432. Modify `PG_CONN` in `download_missing_block.sh` for your environment.
+
+1. Install the required `mina-archive-blocks` and `mina-missing-blocks-auditor` scripts that are packed in the `minaprotocol/mina-archive:1.4.0-c980ba8-bullseye` Docker image.
+
+1. Export the `BLOCKS_BUCKET`:
+
+    ```sh
+    export BLOCKS_BUCKET="https://storage.googleapis.com/my_bucket_with_precomputed_blocks"
+    ```
+
+1. Run the `mina-missing-blocks-auditor` script from the database host:
+    
+    For Devnet:
+    
+    ```sh
+    download-missing-blocks.sh devnet {db_user} {db_password}
+    ```
+
+    For Mainnet:
+    
+    ```sh
+    download-missing-blocks.sh mainnet {db_user} {db_password}
+    ```
+    
+## What steps are required for the Berkeley migration?
```

Comment:
the title of this heading does not seem to match the paragraph below... what are we trying to say here?

Reply:
It's trying to say you need to make sure there's no missing blocks.